requirements-base.txt

```diff
@@ -34,7 +34,7 @@ progressbar2==3.32.0
 # If we bump it to 2.8.4 or greater then Python 3.6 & 3.8 would use the same version
 psycopg2-binary==2.7.7; python_version == '3.6'
 psycopg2-binary==2.8.6; python_version > '3.6'
-PyJWT==1.5.3
+PyJWT==2.1.0
```

Comment:
It seems to me this large refactor is pretty risky. Maybe we should just merge it separately to the also risky version bump?

Reply:
We can fairly easily back out https://github.com/getsentry/sentry/pull/26404 again if that is preferred.  It is why the two were made separately initially.  However the motivation to do as a single one is that:
* the most invasive change is the new wrapper module and adjusting all callers to it
* this needs careful deployment with some manual monitoring etc
* once the wrapper module is adopted the upgrade itself is not so large (you can still see the PR)
* doing it separate means doubling the deploy overhead with manual monitoring for little gain.
* if both are merged and deployed in succession it make reverting more steps and thus more error prone, there are now two PRs to revert rather than one.

But again, we could back this out and merge https://github.com/getsentry/sentry/pull/26404 separately into master if that's preferred.

Reply:
@wedamija @flub, we (ecosystem) has talked it over and here is our plan:

1. Take the upgrading of `PyJWT` out of this PR again and from there have myself, @ceorourke and @mgaeta all do testing of integrations locally to spot check the code refactors
2. Once that's confirmed and tests are passing we will merge and deploy this PR 
3. If any issues occur we will revert 
4. We'll do the same testing for the upgrading lib PR
4. Provided everything is okay with the 1st PR and we've tested the lib upgrade, we'll merge and deploy that one
5. If any issues occur we will revert 

This will take a bit longer than just merging it all as one PR but I think if any issues do occur, we'll know whether it was the refactor or the upgrade itself.